subprojects/common/analysis/src/main/java/hu/bme/mit/theta/analysis/multi/MultiState.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *  Copyright 2024 Budapest University of Technology and Economics
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package hu.bme.mit.theta.analysis.multi;
+
+import com.google.common.base.Objects;
+import hu.bme.mit.theta.analysis.State;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public abstract class MultiState<LState extends State, RState extends State, DataState extends State> implements State {
+
+    private final LState leftState;
+    private final RState rightState;
+    private final DataState dataState;
+
+    /**
+     * Denotes whether this state derived from an action that considered the left state of the previous multi state.
```

Comment:
the doc is no longer true, as it was changed to a more general side definition, not just a bool

Reply:
Resolved in fdc9a9fa3a73f5909d72dd9ff5c9e50b4dda0863